autorest/azure/azure.go

```diff
@@ -29,12 +29,13 @@ const (
 
 // ServiceError encapsulates the error response from an Azure service.
 type ServiceError struct {
-	Code    string `json:"code"`
-	Message string `json:"message"`
+	Code    string       `json:"code"`
+	Message string       `json:"message"`
+	Details *interface{} `json:"details"`
 }
 
 func (se ServiceError) Error() string {
-	return fmt.Sprintf("Azure Error: Code=%q Message=%q", se.Code, se.Message)
+	return fmt.Sprintf("Code=%q Message=%q Details=%v", se.Code, se.Message, se.Details)
```

Comment:
Thought users won't look for details in error when detail is nil. They will know by error string. But got your point that if for RPs that never send details, this will always show up nil. Will update.